src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,2886 @@
+                            for(int k = 0; k < 4; k++)
+                            {
+                                p1[k + 4] = p2[k];
+                                p1[k + 8] = p3[k];
+                            }
+                            rpp_simd_store(rpp_store48_f32pln3_to_u8pln3, dstRowPtrTempR, dstRowPtrTempG, dstRowPtrTempB, p1);    // simd stores
+                            xR += 16;
+                            xG += 16;
+                            xB += 16;
+                        }
+                        else
+                        {
+                            if(xR < roi.xywhROI.roiWidth)
+                            {
+                                for( ; xR < roi.xywhROI.roiWidth; xR++)
+                                {
+                                    *dstRowPtrTempR = *srcRowPtrTempR;
+                                    srcRowPtrTempR += 3;
+                                    dstRowPtrTempR++;
+                                }
+                            }
+                            if(xG < roi.xywhROI.roiWidth)
+                            {
+                                for( ; xG < roi.xywhROI.roiWidth; xG++)
+                                {
+                                   *dstRowPtrTempG = *srcRowPtrTempG;
+                                    srcRowPtrTempG += 3;
+                                    dstRowPtrTempG++;
+                                }
+                            }
+                            if(xB < roi.xywhROI.roiWidth)
+                            {
+                                for( ; xB < roi.xywhROI.roiWidth; xB++)
+                                {
+                                    *dstRowPtrTempB = *srcRowPtrTempB;
+                                    srcRowPtrTempB += 3;
+                                    dstRowPtrTempB++;
+                                }
+                            }
+                            break;
+                        }
+                        srcRowPtrTempR += 48;
+                        srcRowPtrTempG += 48;
+                        srcRowPtrTempB += 48;
+                        dstRowPtrTempR += 16;
+                        dstRowPtrTempG += 16;
+                        dstRowPtrTempB += 16;
+                    }
+                }
+                else
+                {
+                    Rpp8u *srcRowPtrTempR, *srcRowPtrTempG, *srcRowPtrTempB, *dstRowPtrTempR, *dstRowPtrTempG, *dstRowPtrTempB;
+                    srcRowPtrTempR = srcRowPtrR + xR * 3;
+                    srcRowPtrTempG = srcRowPtrG + xG * 3 + 1;
+                    srcRowPtrTempB = srcRowPtrB + xB * 3 + 2;
+                    dstRowPtrTempR = dstRowPtr;
+                    dstRowPtrTempG = dstRowPtr + dstDescPtr->strides.cStride;
+                    dstRowPtrTempB = dstRowPtr + 2 * dstDescPtr->strides.cStride;
+                    if(yR < roi.xywhROI.roiHeight && xR < roi.xywhROI.roiWidth)
```

Comment:
@HazarathKumarM Why do we need so many if statements between L160-L186? Even if we need it, there is so much repeated code @sampath1117 please check.